core/src/main/scala/org/apache/spark/sql/rapids/tool/ToolUtils.scala

```diff
@@ -444,11 +446,17 @@ object ExecHelper {
     ExecuteRefreshTable,
     ExecuteRepairTableCommand,
     ExecuteShowPartitionsCommand,
+    ExecuteClearCacheCommand,
+    ExecuteOptimizeTableCommandEdge,
     SubqueryExecParser.execName
   )
 
   def shouldIgnore(execName: String): Boolean = {
-    getAllIgnoreExecs.contains(execName)
+    // Normalize the execName by removing the trailing '$' character, if present.
+    // This is necessary because in Scala, the '$' character is often appended to the names of
+    // generated classes or objects, and we want to match the base name regardless of this suffix.
+    val normalizedExecName = execName.stripSuffix("$")
+    getAllIgnoreExecs.contains(normalizedExecName)
```

Comment:
QQ: Should we do this stripping somewhere else when we process a node?
Otherwise, the name stored in the AppInfo will still have the trailing `$`

Reply:
Thanks @amahussein! Moved this to parsePlanNode so that Exec name doesn't contain trailing `$` in AppInfo and in output csv files. PTAL.